classes/checkout/CheckoutAddressesStep.php

```diff
@@ -202,51 +214,77 @@ public function handleRequest(array $requestParams = array())
 
     public function getTemplateParameters()
     {
-        $idAddressDelivery = (int) $this->getCheckoutSession()->getIdAddressDelivery();
-        $idAddressInvoice = (int) $this->getCheckoutSession()->getIdAddressInvoice();
-        $addressWarning = false;
-        $errorKey = 'address_error';
-        $checkoutWarning = isset($this->context->controller->checkoutWarning['address'])
-            ? $this->context->controller->checkoutWarning['address']
+        /** @var OrderControllerCore $controller */
+        $controller = $this->context->controller;
+        $warnings   = $controller->checkoutWarning;
+
+        $idAddressDelivery = (int)$this->getCheckoutSession()->getIdAddressDelivery();
+        $idAddressInvoice  = (int)$this->getCheckoutSession()->getIdAddressInvoice();
+        $addressWarning    = isset($warnings['address'])
+            ? $warnings['address']
             : false;
-        $invalidAddresses = isset($this->context->controller->checkoutWarning['invalid_addresses'])
-            ? $this->context->controller->checkoutWarning['invalid_addresses']
+        $invalidAddresses  = isset($warnings['invalid_addresses'])
+            ? $warnings['invalid_addresses']
             : array();
 
-        if (!empty($checkoutWarning)) {
-            if ($idAddressDelivery === (int)$checkoutWarning['id_address']) {
-                $addressWarning = $checkoutWarning;
-                $errorKey = 'delivery_address_error';
-            } elseif ($idAddressInvoice === (int)$checkoutWarning['id_address']) {
-                $addressWarning = $checkoutWarning;
-                $errorKey = 'invoice_address_error';
-                $this->use_same_address = false;
```

Comment:
By refactoring this specific line, you made the code less logical. We must not forget that the key 'invoice_address_error' has now consequences in the next test.

Reply:
Indeed. We needed both keys (`delivery_address_error` and `invoice_address_error`) to exist at the same time when relevant.